lib/active_merchant/billing/gateways/iats_payments.rb

```diff
@@ -42,7 +43,7 @@ def purchase(money, payment, options={})
         add_ip(post, options)
         add_description(post, options)
 
-        commit((payment.is_a?(Check) ? :purchase_check : :purchase), post)
+        commit((payment.is_a?(Check) ? :purchase_check : payment.is_a?(String) ? :purchase_token : :purchase), post)
```

Comment:
Nested ternary operators 🙅‍♂️  could you please change this to something similar to the structure used in `add_payment`?